api/helpers/lock-game.js

```diff
@@ -34,15 +34,22 @@ module.exports = {
         const now = dayjs.utc().toDate();
         const lockIsStaleTimeout = dayjs.utc().subtract(30, 'second').toDate();
 
-        // Lock & re-fetch game if unlocked or lock is expired
-        const updatedGame = await Game.updateOne({
+        // Lock game if unlocked or lock is expired
+        let updatedGame = await Game.updateOne({
           id: gameId,
           or: [{ lock: null }, { lockedAt: { '<=': lockIsStaleTimeout } }],
         }).set({ lock: uuId, lockedAt: now });
 
         // If we successfully wrote our uuid, resolve
         const newLock = updatedGame?.lock;
         if (newLock === uuId) {
+
+          // Re-fetch game and populate players and gamestates
+          updatedGame = await Game.findOne({ id: gameId })
+            .populate('p0')
+            .populate('p1')
+            .populate('gameStates', { sort: 'createdAt ASC' });
```

Comment:
Updating `lockGame()` to include these associations so that consumers can exclusively rely on it for game queries instead of having to requery for associations after locking the game. Unfortunately it still requires an extra find query (in addition the `Game.updateOne()` above which also fetches after update), but at least that's abstracted into this helper.

The upshot is that endpoints that need to grab an individual game can lock it and have all the data they need to complete the request